src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -86,21 +100,33 @@ public static CollectorManager createHybridCollectorManager(final SearchContext
                 isSingleShard,
                 trackTotalHitsUpTo,
                 searchContext.sort(),
-                filteringWeight
+                filteringWeight,
+                searchContext.searchAfter()
             )
             : new HybridCollectorNonConcurrentManager(
                 numDocs,
                 new HitsThresholdChecker(Math.max(numDocs, searchContext.trackTotalHitsUpTo())),
                 isSingleShard,
                 trackTotalHitsUpTo,
                 searchContext.sort(),
-                filteringWeight
+                filteringWeight,
+                searchContext.searchAfter()
             );
     }
 
     @Override
     public Collector newCollector() {
-        Collector hybridcollector = new HybridTopScoreDocCollector(numHits, hitsThresholdChecker);
+        Collector hybridcollector;
+        if (sortAndFormats != null) {
+            if (after == null) {
+                hybridcollector = new SimpleFieldCollector(numHits, hitsThresholdChecker, sortAndFormats.sort);
+            } else {
+                validateSearchAfterFieldAndSortFormats();
+                hybridcollector = new PagingFieldCollector(numHits, hitsThresholdChecker, sortAndFormats.sort, after);
+            }
```

Comment:
do we support pagination currently with Hybrid Query clause? My understanding was no.

Reply:
No we do not. This `search_after` works exclusively with sorting. I have added this functionality with this feature so that in future I do not have to touch this class again for adding any new functionality. Next thing in my roadmap is enable pagination in hybrid search. https://opensearch.org/docs/latest/search-plugins/searching-data/paginate/#the-search_after-parameter

Reply:
I would not code for pagination in this iteration, mainly to make complexity of this change less. If that's a reasonable amount of effort can be remove pagination specific code? We can add it together with the rest of changes for pagination

Reply:
Consider in future when we roll out pagination feature then there are 4 types of different pagination. All cannot go into the same PR as they are altogether different as a individual. Now, the thing with `search_after` it is widely used with sorting, that is why its code looks almost same with simpleFieldCollector. Later, when we roll out pagination we need not have to touch sorting again for this purpose. That is the reason I am rolling `search_after` feature with this PR.

Reply:
Discussed offline with @martin-gaievski and we are keeping this collector in this PR.

Reply:
To share the context - wording in previous comment need to be reviewed, this is part of the sorting feature, fac that it can be re-used in "pagination" is just a nice bonus, not a goal